### DISCUSSION TITLE
How to get onColumnHeaderClicked event in aggrid component

### DISCUSSION THREAD

**Discussion by Husky0c** [zauberzeug/nicegui#3801] [Q&A] [answered]
I call the click event through ui.aggrid().on('CellClicked'), and it works well, I get the event object. But when I use .on('onColumnHeaderClicked'), things go wrong, and nothing happens when I click the ag-Grid header. Did I get the name wrong, or does NiceGUI not support listening to the ColumnHeaderClicked event? Thx to all.

**Comment by python-and-novella** [comment]
Plz provide a minimal reproduce example.

**Comment by python-and-novella** [comment, marked as answer]
```python3
from nicegui import ui

ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
    ],
}).on('ColumnHeaderClicked', lambda event: ui.notify(f'ColId: {event.args['colId']}'))

ui.run()
```

**Reply by Husky0c** [reply]
Wow, that's awesome. Thank you very much for providing an example. I realized that you are using 'ColumnHeaderClicked' and that I don't need to add 'on' in front of it. Anyway, thanks again.

**Reply by Anurag-Singh-Yadav** [reply]
I am using this in react, could you please help me how to add ColumnHeaderClicked on column
